### PR TITLE
chore: add client-side validation for email and picture url fields

### LINKED ISSUE #5855: add client side form field validation for email and profile picture
### Current Behavior

In the community member form (https://layer5.io/newcomers):

1. The `Picture` field should accept a url to an image and not a data (base64 encoded, etc.) file.
2. Email validation is broken.
### Desired Situation

The above issues should be fixed

---
<h2>Contributor Resources and <a href="https://layer5.io/community/handbook">Handbook</a></h2>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- 📚 See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md).
- 🎨 Wireframes and [designs for Layer5 site](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)

- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.layer5.io).

Join the Layer5 Community by submitting your [community member form](https://layer5.io/newcomer).

### PR DESCRIPTION
**Pull request by vishalvivekm**
**Description**

This PR fixes #5855

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit fbbebe76955f6aab3a555885d0060527e5fcea20 at: https://66d463d25429312a1edc6549--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 3c974b5ac195441610d864e5eb2b43e5c018e4c1 at: https://66d469aab15368323b37f867--layer5.netlify.app

File: src/sections/Community/Web-based-from/index.js [hunk 1]
```diff
@@ -7,6 +7,38 @@ import { Field, Formik, Form } from "formik";
 import axios from "axios";
 import { Link } from "gatsby";
 
+const validateEmail = (value) => {
+  let error;
+
+  if (!value) {
+    error = "Required";
+  } else if (!/^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}$/.test(value)) {
```
Comment (leecalcote) [inline src/sections/Community/Web-based-from/index.js:15]: Is this the same regex used by the other site forms?

File: src/sections/Community/Web-based-from/index.js [hunk 2]
```diff
@@ -7,6 +7,38 @@ import { Field, Formik, Form } from "formik";
...
+  }
+  return error;
+};
+
+const validatePictureUrl = (value) => {
+  let error;
+  if (value) {
+
+    if (value.startsWith("data:")) {
+      error = "Data URIs are not allowed. Please provide a URL to an image file.";
```
Comment (leecalcote) [inline src/sections/Community/Web-based-from/index.js:26]:
```suggestion
      error = "Data URIs are not allowed. Please provide a URL, starting with http:// or https:// to an image file.";
```

File: src/sections/Community/Web-based-from/index.js [hunk 3]
```diff
@@ -7,6 +7,38 @@ import { Field, Formik, Form } from "formik";
...
+    } else {
+      try {
+        new URL(value);
+        const allowedImageExtensions = ["jpg", "jpeg", "png", "webp", "svg"];
+        const extension = value.split(".").pop().toLowerCase();
+        if (!allowedImageExtensions.includes(extension)) {
+          error = "URL must point to an image file (jpg, jpeg, png, svg or webp).";
```
Comment (leecalcote) [inline src/sections/Community/Web-based-from/index.js:33]: Not a gif?

File: src/sections/Community/Web-based-from/index.js [hunk 4]
```diff
@@ -7,6 +7,38 @@ import { Field, Formik, Form } from "formik";
...
+        }
+      } catch (_) {
+        error = "Please enter a valid URL.";
```
Comment (leecalcote) [inline src/sections/Community/Web-based-from/index.js:36]:
```suggestion
        error = "Please enter a URL to an image file.";
```

File: src/sections/Community/Web-based-from/index.js (hunk 1 above)
Comment (vishalvivekm) [reply src/sections/Community/Web-based-from/index.js:15]: Yes.

File: src/sections/Community/Web-based-from/index.js [hunk 5]
```diff
@@ -7,6 +7,38 @@ import { Field, Formik, Form } from "formik";
...
+const validatePictureUrl = (value) => {
+  let error;
+  if (value) {
+
+    if (value.startsWith("data:")) {
+      error = "Data URIs are not allowed. Please provide a URL to an image file.";
+    } else {
+      try {
+        new URL(value);
+        const allowedImageExtensions = ["jpg", "jpeg", "png", "webp", "svg"];
```
Comment (vishalvivekm) [inline src/sections/Community/Web-based-from/index.js:30]:
```suggestion
        const allowedImageExtensions = ["jpg", "jpeg", "png", "webp", "svg", "gif"];
```

File: src/sections/Community/Web-based-from/index.js (hunk 3 above)
Comment (vishalvivekm) [inline src/sections/Community/Web-based-from/index.js:33]:
```suggestion
          error = "URL must point to an image file (jpg, jpeg, png, svg, webp or gif).";
```

Comment (vishalvivekm) [reply src/sections/Community/Web-based-from/index.js]: Added

**Comment by l5io** [comment]
🚀 Preview for commit 95596dd2d7541f0f1f219594c60f33e998b5f59e at: https://66d47095498964065614936e--layer5.netlify.app

**Review by sudhanshutech — APPROVED** [review]

**Pull request merged**